exchange/utils.go

```diff
@@ -847,6 +854,60 @@ func mergeBidderRequests(allBidderRequests []BidderRequest, bidderNameToBidderRe
 	return allBidderRequests
 }
 
+func setLegacyGDPRFromGPP(r *openrtb2.BidRequest, gpp gpplib.GppContainer) {
+	if r.Regs != nil && r.Regs.GDPR == nil {
+		if r.Regs.GPPSID != nil {
+			// Set to 0 unless SID exists
+			r.Regs.GDPR = ptrutil.ToPtr[int8](0)
+			for _, id := range r.Regs.GPPSID {
+				if id == int8(gppConstants.SectionTCFEU2) {
+					regs := *r.Regs
+					regs.GDPR = ptrutil.ToPtr[int8](1)
+					r.Regs = &regs
+				}
+			}
+		}
+	}
+
+	if r.User == nil || len(r.User.Consent) == 0 {
+		for i, id := range gpp.SectionTypes {
+			if id == gppConstants.SectionTCFEU2 {
+				if r.User == nil {
+					r.User = &openrtb2.User{
+						Consent: gpp.Sections[i].GetValue(),
+					}
+				}
+				user := *r.User
+				user.Consent = gpp.Sections[i].GetValue()
+				r.User = &user
```

Comment:
Consider using an else clause to avoid setting the consent twice and making an unnecessary copy:

```
if r.User == nil {
	r.User = &openrtb2.User{
		Consent: gpp.Sections[i].GetValue(),
	}
} else {
	user := *r.User
	user.Consent = gpp.Sections[i].GetValue()
	r.User = &user
}
```

Reply:
LOL. No worries.